src/Application/Migrations/24_function_ticket_reset.sql

```diff
@@ -0,0 +1,70 @@
+BEGIN;
+
+--param_handle_id is used for the log entry
+CREATE OR REPLACE FUNCTION ticket_reset(param_ticket_id bigint, param_handle_id bigint)
+  RETURNS void AS
+$$
+DECLARE
+  ticket tbl_ticket%rowtype;
+  t tbl_ticket%rowtype;
+  inital_state enum_ticket_state;
+BEGIN
+  --first get the full given ticket
+  SELECT * INTO ticket FROM tbl_ticket where id = param_ticket_id;
+
+  IF ticket IS NULL THEN
+    RAISE WARNING 'ticket with id % does not exist', param_ticket_id;
+    RETURN;
+  END IF;
+
+  --reset the given ticket to inital state
+  inital_state := ticket_state_initial(ticket.project_id, ticket.ticket_type);
+  IF inital_state <> ticket.ticket_state THEN
+    INSERT INTO tbl_log (
+      ticket_id, from_state, to_state, handle_id, event
+      ) VALUES (ticket.id, ticket.ticket_state, inital_state, param_handle_id, 'Action.reset');
+    UPDATE tbl_ticket SET ticket_state=inital_state WHERE id=param_ticket_id;
+  END IF;
+
+  --if given ticket is a meta ticket, reset all childs
+  IF ticket.ticket_type = 'meta'::enum_ticket_type THEN
+    FOR t IN
+    SELECT * FROM tbl_ticket WHERE parent_id=param_ticket_id
+    LOOP
+      inital_state := ticket_state_initial(t.project_id, t.ticket_type);
+      IF inital_state <> t.ticket_state THEN
+        INSERT INTO tbl_log (
+          ticket_id, from_state, to_state, handle_id, event, comment
+          ) VALUES (t.id, t.ticket_state, inital_state, param_handle_id, 'Action.reset', 'Following meta Ticket.');
+        UPDATE tbl_ticket SET ticket_state=inital_state WHERE id=t.id;
+      END IF;
+    END LOOP;
+
+  --if given ticket is an encoding ticket , reset all ticket depends on it by encoding profile
+  ELSEIF ticket.ticket_type = 'encoding'::enum_ticket_type THEN
```

Comment:
An here I'd rather just check, if ticket.encoding_profile_version_id IS NOT NULL, since that's the important column in the join clause, but technically you could even do this query conditionally too. For other ticket types this loop should do nothing.